app/models/post.rb

```diff
@@ -1,3 +1,21 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_author_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ OPTIONAL ] I will suggest that you consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) : `after_save` to call the `#update_author_posts_counter` method every time a new post has been saved to database, then make the `update_author_posts_counter` method private. 

- Example Implementation
```ruby
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```